pkg/controller/nodedaemon/reconcile.go

```diff
@@ -41,6 +51,12 @@ type DaemonReconciler struct {
 func (r *DaemonReconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
 	r.reqLogger = logf.Log.WithName(controllerName).WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
 
+	// do a one-time delete of the old static-provisioner daemonset
+	err := r.cleanupOldDaemonsets(request.Namespace)
```

Comment:
Can you confirm if we log an event to `LocalVolumeSet` object if deleting this daemonset fails? The idea is - if somehow the `LocalVolumeSet` rollout fails after LSO upgrade, I wouldn't want customer to dig through logs to find out what happened and hence an event would be appropriate place to log this.

Reply:
Currently, an event reporter is not implemented in the main controller for localvolumeset. We have a refactoring task open for exporting a common event reporter package. How about we add this one once the common event reporter package is in?

I agree that any long-lasting failure should bubble up as an event, but I don't see that happening here.

Reply:
Anyway, this is my idea for centralizing eventing: https://github.com/openshift/local-storage-operator/pull/204/files

Reply:
> I agree that any long-lasting failure should bubble up as an event, but I don't see that happening here.

I do not understand this. Customer had a working LSO in 4.6 and then they upgrade to 4.8 and LSO is stuck now. We should have clear reporting for this. Are you saying - there is no chance that DS clean up fails? There are plenty of reasons that DS cleanup may fail. For example - if any of the nodes where daemonset was running is unresponsive, then pod deletion on that node may get stuck. IMO - there are other failure scenarios too. The least we could do is, report proper errors when this happens so as customers know what to do.